app/gui/src/controller/graph/executed.rs

```diff
@@ -216,28 +216,30 @@ impl Handle {
         registry.clone_ref().get_type(id)
     }
 
-    /// Enter node by given node ID and method pointer.
+    /// Enter the given stack of nodes by node ID and method pointer.
     ///
-    /// This will cause pushing a new stack frame to the execution context and changing the graph
-    /// controller to point to a new definition.
+    /// This will push new stack frames to the execution context and change the graph controller to
+    /// point to a new definition.
     ///
     /// ### Errors
     /// - Fails if method graph cannot be created (see `graph_for_method` documentation).
     /// - Fails if the project is in read-only mode.
-    pub async fn enter_method_pointer(&self, local_call: &LocalCall) -> FallibleResult {
+    pub async fn enter_stack(&self, stack: Vec<LocalCall>) -> FallibleResult {
         if self.project.read_only() {
             Err(ReadOnly.into())
         } else {
-            debug!("Entering node {}.", local_call.call);
-            let method_ptr = &local_call.definition;
-            let graph = controller::Graph::new_method(&self.project, method_ptr);
-            let graph = graph.await?;
-            self.execution_ctx.push(local_call.clone()).await?;
-            debug!("Replacing graph with {graph:?}.");
-            self.graph.replace(graph);
-            debug!("Sending graph invalidation signal.");
-            self.notifier.publish(Notification::EnteredNode(local_call.clone())).await;
-
+            for local_call in stack.clone() {
```

Comment:
Why do we need to clone the stack here?

Reply:
Because the stack is also passed on later later in this function to the notification back to the view (so we know which breadcrumbs to add).